hyperspy/signal.py

```diff
@@ -3649,10 +3648,11 @@ def as_signal1D(self, spectral_axis, out=None):
         <Signal1D, title: , dimensions: (6, 5, 3, 4)>
 
         """
-        # Roll the spectral axis to-be to the latex index in the array
-        sp = self.rollaxis(spectral_axis, -1 + 3j)
-        sp.axes_manager.set_signal_dimension(1)
-        sp._assign_subclass()
+        ax = self.axes_manager[spectral_axis]
+        all_ax = self.axes_manager._axes.copy()
+        sig = [all_ax.pop(ax.index_in_array)]
+        sp = self.transpose(signal_axes=sig,
```

Comment:
Wouldn't `self.transpose(signal_axes=self.navigation_axes[-1], copy=True)` be equivalent?